.github/workflows/publish_test.yml

```diff
@@ -0,0 +1,50 @@
+# This github action builds Alibi Detect and publishes it to Test PyPI.
+# It is only run when a new tag is pushed to github.
+name: Publish release to Test PyPI
+
+# Only run when tags starting with "v" are pushed
+on: 
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build-and-publish:
+    name: Build and publish
+    runs-on: ubuntu-18.04
+
+    steps:
+    # Setup Python
+    - uses: actions/checkout@master
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.9
+
+    # Build
+    - name: Build a binary wheel and a source tarball
+      run: make build
+
+    # Publish to Test PyPI
+    - name: Publish 📦 to Test PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+
+    # Get the PEP440 compliant version number (so we can check version is installable)
+    - name: Get 📦 version number
+      id: get_version
+      run: echo "PACKAGE_VERS=$(python setup.py --version)" >> $GITHUB_ENV
+
+    # Wait for a while to give PyPI time to update (otherwise the pip install might fail)
+    - name: Sleep for 10 minutes
+      shell: bash
+      run: sleep 10m
+
+    # Install from Test PyPI and test
+    - name: Install 📦 from PyPI and test
+      run: |
+        make clean
+        python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple alibi-detect==${{ env.PACKAGE_VERS }}
+        python -c "from alibi_detect.cd import KSDrift"
```

Comment:
I would change this to have imports like `import alibi_detect`, `from alibi_detect.cd import *` etc. Also, does it  capture any stdout/stderr and print it out? Asking as I didn't see it in the example CI run here: https://github.com/ascillitoe/alibi-detect/actions/runs/3490747918/jobs/5842565978

Reply:
Re stdout I think this line was stdout:

```
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
```

In https://github.com/SeldonIO/alibi-detect/pull/674/commits/b3aa94ef624ec5e068841742f83069d0f06de940 I've changed to the imports you suggested and added `print()`'s so there is more obvious stdout. CI result [here](https://github.com/ascillitoe/alibi-detect/actions/runs/3496483248/jobs/5854443831).

Reply:
Re stderr I'm not sure a python error would actually fail the GA step. Will look into this. Think we're pretty much there after that!

Reply:
OK seems to fail as desired:
https://github.com/ascillitoe/alibi-detect/actions/runs/3496723210/jobs/5854969140